storage/replica.go

```diff
@@ -3376,6 +3377,59 @@ func (r *Replica) maybeGossipSystemConfig() {
 	r.systemDBHash = hash
 }
 
+// maybeGossipNodeLiveness gossips information for all node liveness
+// records stored on this range. To scan and gossip, this replica
+// must hold the lease to a range which contains some or all of the
+// node liveness records. After scanning the records, it checks
+// against what's already in gossip and only gossips records which
+// are out of date.
```

Comment:
Gossip.AddInfo could check and see if the new value is already present, and if so do nothing.